lms/static/js/spec/bookmarks/bookmarks_spec.js

```diff
@@ -0,0 +1,134 @@
+                };
+
+                it("has correct behavior for bookmarks button", function () {
+                    spyOn(bookmarksButtonView, 'toggleBookmarksListView').andCallThrough();
+                    spyOn(bookmarksButtonView.bookmarksListView, 'showBookmarksList').andReturn(true);
+
+                    bookmarksButtonView.delegateEvents();
+
+                    expect(bookmarksButtonView.$('.bookmarks-list-button')).toHaveAttr('aria-pressed', 'false');
+                    expect(bookmarksButtonView.$('.bookmarks-list-button')).toHaveClass('is-inactive');
+
+                    bookmarksButtonView.$('.bookmarks-list-button').click();
+                    expect(bookmarksButtonView.toggleBookmarksListView).toHaveBeenCalled();
+                    expect(bookmarksButtonView.$('.bookmarks-list-button')).toHaveAttr('aria-pressed', 'true');
+                    expect(bookmarksButtonView.$('.bookmarks-list-button')).toHaveClass('is-active');
+
+                    bookmarksButtonView.$('.bookmarks-list-button').click();
+                    expect(bookmarksButtonView.$('.bookmarks-list-button')).toHaveAttr('aria-pressed', 'false');
+                    expect(bookmarksButtonView.$('.bookmarks-list-button')).toHaveClass('is-inactive');
+                });
+
+                it("has rendered bookmarked list correctly", function () {
+                    var requests = AjaxHelpers.requests(this);
+                    var url = bookmarksButtonView.bookmarksListView.url + '?course_id=COURSE_ID&fields=path';
+                    var expectedData = createBookmarksData();
+                    var bookmarksListView = bookmarksButtonView.bookmarksListView;
+
+                    spyOn(bookmarksListView, 'courseId').andReturn('COURSE_ID');
+                    bookmarksButtonView.$('.bookmarks-list-button').click();
+
+                    expect($('#loading-message').text().trim()).toBe(bookmarksListView.loadingMessage);
+
+                    AjaxHelpers.expectRequest(requests, 'GET', url);
+                    AjaxHelpers.respondWithJson(requests, expectedData);
+
+                    expect(bookmarksListView.$('.bookmarks-results-header').text().trim()).toBe('My Bookmarks');
+
+                    verifyBookmarkedData(bookmarksListView, expectedData);
+                });
+
+                it("can navigate to correct url", function () {
+                    var requests = AjaxHelpers.requests(this);
+                    var bookmarksListView = bookmarksButtonView.bookmarksListView;
+
+                    spyOn(bookmarksListView, 'visitBookmark');
+
+                    bookmarksButtonView.$('.bookmarks-list-button').click();
+
+                    AjaxHelpers.respondWithJson(requests, createBookmarksData());
+
+                    bookmarksListView.$('.bookmarks-results-list-item').click();
+                    expect(bookmarksListView.visitBookmark).toHaveBeenCalled();
```

Comment:
I think it would be good to verify that the browser is directed to the correct url.